crates/bevy_ecs/src/component.rs

```diff
@@ -1098,6 +1098,20 @@ pub struct Components {
 }
 
 impl Components {
+    /// Generates a new [`ComponentId`] for the provided Rust type from its [`TypeId`],
+    /// returning any existing [`ComponentId`] if the type has already been registered.
+    ///
+    /// For read-only access, see [`Components::component_id`].
+    pub fn generate_component_id<T: ?Sized + 'static>(&mut self) -> ComponentId {
+        if let Some(component_id) = self.indices.get(&TypeId::of::<T>()) {
+            component_id.clone()
+        } else {
+            let id = ComponentId(self.components.len());
```

Comment:
If we aren't registering a component info won't this return the same ID multiple times since the length of self.components does not increase?